File: 2331-evaluate-boolean-binary-tree/2331-evaluate-boolean-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean evaluateTree(TreeNode root) {
        return order(root);
    }
    
    public boolean order(TreeNode node){
        
        if(node.left == null && node.right == null){
            if(node.val == 1){
                return true;
            }
            return false;
        }
        boolean left = order(node.left);
        boolean right = order(node.right);
        
        if(node.val == 2){
            return left || right;
        }else{
            return left && right;
        }
    }
    
}

File: 2331-evaluate-boolean-binary-tree/2331-evaluate-boolean-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def evaluateTree(self, root: Optional[TreeNode]) -> bool:
        
        
        def order(node):
            
            if node.left is None and node.right is None:
                if node.val == 1:
                    return True
                return False
            
            x = order(node.left)
            y = order(node.right)
            
            if node.val == 2:
                return x or y
            else:
                return x and y
            
        
        return order(root)
            
        
        